02_bubble/brnbke/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(arr []int) []int {
+	changed := true
+	r := make([]int, len(arr))
+	copy(r, arr)
+
+	for {
+		if !changed {
+			break
+		}
+		changed = false
+		for i := 0; i < len(r)-1; i++ {
+			if r[i] > r[i+1] {
+				r[i], r[i+1] = r[i+1], r[i]
+				changed = true
+			}
+		}
+	}
+
+	return r
+}
+
+func insertion(arr []int) []int {
+	result := make([]int, len(arr))
+	copy(result, arr)
+
+	for i := 1; i < len(result); i++ {
+		tmp := result[i]
+		for j := i - 1; j > -1; j-- {
```

Comment:
nit: `j > -1` is a little odd. I'd use `j >= 0`, since 0 has meaning but -1 does not (except as being 1 off the meaningful number).

Reply:
Implemented.

@samuong seems you were not being too nitpicky with this suggestion either.

Reply:
Hah, good to know I'm not alone! 😄